R/tidyverse.R

```diff
@@ -101,6 +104,37 @@ use_tidy_description <- function() {
   invisible(TRUE)
 }
 
+#' @export
+#' @rdname tidyverse
+use_tidy_dependencies <- function() {
+  use_dependency("rlang", "Imports")
+  use_dependency("ellipsis", "Imports")
+  use_dependency("lifecycle", "Imports")
+  use_dependency("cli", "Imports")
+  use_dependency("glue", "Imports")
+  use_dependency("withr", "Imports")
+
+  # add badges; we don't need the details
+  ui_silence(use_lifecycle())
+
+  # standard imports
+  imports <- roxygen_ns_append("@import rlang") ||
+    roxygen_ns_append("@importFrom glue glue")
+  if (imports) {
+    roxygen_update_ns(TRUE)
+  }
+
+  # If needed, copy in lightweight purrr compatibility layer
+  if (!desc::desc(proj_get())$has_dep("purrr")) {
+    url <- "https://raw.githubusercontent.com/r-lib/rlang/master/R/compat-purrr.R"
+    path <- tempfile()
+    utils::download.file(url, path, quiet = TRUE)
+    write_over("R/compat-purrr.R", readLines(path))
```

Comment:
Just for internal consistency?
```suggestion
    write_over("R/compat-purrr.R", read_utf8(path))
```

Reply:
This is why we need a linter 🤪